### PR TITLE
SCSS converter: expand @config["source"] to be "safer".

### PR DESCRIPTION
**Pull request by parkr**
Addresses a potential issue from #50.

/cc @jekyll/gh-pages

### CODE REVIEW

File: lib/jekyll/converters/scss.rb [hunk 1]
```diff
@@ -122,6 +122,11 @@ def convert(content)
       rescue ::Sass::SyntaxError => e
         raise SyntaxError.new("#{e.to_s} on line #{e.sass_line}")
       end
+
+      private
+      def site_source
+        @site_source ||= File.expand_path(@config["source"]).freeze
```
Comment (benbalter) [inline lib/jekyll/converters/scss.rb]: > Jekyll.sanitized_path does not expand the base directory, so we have to do that here.

Would the smarter fix be to do that there?

File: lib/jekyll/converters/scss.rb [hunk 2]
```diff
@@ -71,25 +71,25 @@ def user_sass_load_paths
...
 
         if safe?
           # Sanitize paths to prevent any attack vectors (.e.g. `/**/*`)
-          paths.map! { |path| Jekyll.sanitized_path(@config["source"], path) }
+          paths.map! { |path| Jekyll.sanitized_path(site_source, path) }
         end
 
         # Expand file globs (e.g. `node_modules/*/node_modules` )
-        Dir.chdir(@config["source"]) do
+        Dir.chdir(site_source) do
```
Comment (benbalter) [inline lib/jekyll/converters/scss.rb]: Does this need `Jekyll.sanitized_path`?

File: lib/jekyll/converters/scss.rb (hunk 1 above)
Comment (parkr) [inline lib/jekyll/converters/scss.rb]: On the one hand, that would mean you can pass any directory and it'll auto-expand, on the other `File.expand_path` calls are pretty expensive and we use this function a *lot*.

Comment (parkr) [inline lib/jekyll/converters/scss.rb]: It'd also lock us to a particular Jekyll version.

**Comment by parkr** [comment]
@jekyllbot: merge +bug

**Pull request merged**